### PR TITLE
Deprecate AWS IOs (Java) using AWS SDK v1 in favor of IOs in amazon-web-services2

### LINKED ISSUE #21249: Deprecate amazon-web-services & kinesis module (SDK v1)
As discussed with [~aromanenko] and previously in the community, goal is to keep only IOs for AWS SDK v2. The module for AWS SDK v1 will be deprecated and eventually be removed to avoid maintaining two code basis.

As part of this ticket we need to evaluate if there's potentially missing features in aws2 that would prevent deprecation / removal.

Particularly lack of writes for KinesisIO in aws2 are a known gap and have to be addressed.

Imported from Jira [BEAM-13174](https://issues.apache.org/jira/browse/BEAM-13174). Original Jira may contain additional context.
Reported by: mosche.

### PR DESCRIPTION
**Pull request by mosche**
Currently there's two different versions of AWS IO connectors in Beam for the Java SDK:

* [amazon-web-services](https://github.com/apache/beam/tree/master/sdks/java/io/amazon-web-services) and [kinesis](https://github.com/apache/beam/tree/master/sdks/java/io/kinesis) for the AWS Java SDK v1
* [amazon-web-services2](https://github.com/apache/beam/tree/master/sdks/java/io/amazon-web-services2) (including kinesis) for the AWS Java SDK v2

This deprecates IOs in both, the  `amazon-web-services` and `kinesis` module in favor of the `amazon-web-services2` module and closes #21249.

After a few Beam releases both, the `amazon-web-services` and `kinesis` module will be removed.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by mosche** [comment]
R: @aromanenko-dev 
R: @aaltay

File: sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/s3/S3FileSystem.java [hunk 1]
```diff
@@ -120,6 +124,10 @@ class S3FileSystem extends FileSystem<S3ResourceId> {
         MoreExecutors.listeningDecorator(
             Executors.newFixedThreadPool(
                 config.getS3ThreadPoolSize(), new ThreadFactoryBuilder().setDaemon(true).build()));
+
+    LOG.warn(
```
Comment (aromanenko-dev) [inline sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/s3/S3FileSystem.java:128]: I'd suggest to add similar warning log message to all other deprecated IO `expand` or read/write methods for more visibility of this change.

**Review by aromanenko-dev — APPROVED** [review]
Thanks! Long time expected deprecation since it's a burden to support two similar modules.

LGTM, just a one suggestion.

**Comment by aromanenko-dev** [comment]
retest this please

**Comment by aromanenko-dev** [comment]
Run Java PreCommit

File: sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/s3/S3FileSystem.java (hunk 1 above)
Comment (mosche) [reply sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/s3/S3FileSystem.java:128]: The reason I added it here is that filesystems are loaded automatically... for all other IOs users should see a warning in their IDE / when compiling right? That feels a lot more noticeable than the log messages.

BTW, here's the reason the logs for S3FileSystem will be extremely verbose :/
Looks like there's a severe problem regarding initialisation of file systems with respect to `SerializablePipelineOptions`, though didn't get any feedback 
https://lists.apache.org/thread/sp9g643lfyj6fzg9m2pym23kjv6zkbmq

**Comment by mosche** [comment]
retest this please

File: sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/s3/S3FileSystem.java (hunk 1 above)
Comment (mosche) [reply sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/s3/S3FileSystem.java:128]: ✔️ done

**Comment by mosche** [comment]
retest this please

**Pull request merged**